### PR TITLE
Creates filter for volunteer table on dashboard

### PR DESCRIPTION
**Pull request by colinsoleim**
Resolves issues #99, #98, #97, and #93

### Description
- Adds jquery datatables for easy filtering and sorting on volunteers table
- Adds draper gem to organize presentation logic in decorators
- Adds basic bootstrap styles to all tables on dashboard
- Adds user.supervisor relationship using supervisor_volunteers table

### New Dependencies
This PR adds two dependencies:
- The first is jquery datatables (https://datatables.net/examples/plug-ins/range_filtering.html) which is a javascript library that gives an easy way to sort and filter HTML tables. I was planning to put this on all of the dashboard tables because of how nice the default sorting is, but specifically I want to use it for the volunteers table to enable filtering by active/inactive status and supervisor.
- The second is draper (https://github.com/drapergem/draper) which is a gem that sets up decorator files which allow you to format data specifically for the view layer without having to clog up the models or the views themselves. 

### Type of change
* New feature (non-breaking change which adds functionality)

### How will this affect user permissions?
No impact.

### How Has This Been Tested?
I've tested this locally and all existing rspec tests pass. I've also added several new rspec tests, one for the filtering feature and several for the decorators.

### Screenshots
Screen Shot 2020-04-18 at 10 03 22 PM

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -3,4 +3,4 @@ require:
   - rubocop-rspec
 
 Style/Documentation:
-  Enforce: False
+  Enabled: False
```
Comment (compwron) [inline .rubocop.yml:6]: Oops, thanks :)

**Comment by colinsoleim** [comment]
This PR is now ready for review. I'd be happy to refactor if people have strong feelings against the draper gem. I also made a lot of changes to the `.travis.yml` file to compile webpack and run Chrome in headless mode for javascript enabled tests, so people might want to review that closely as well.

File: app/controllers/dashboard_controller.rb [hunk 2]
```diff
@@ -4,8 +4,8 @@ class DashboardController < ApplicationController
   def show
     authorize :dashboard
 
-    @volunteers = policy_scope(User.volunteer)
+    @volunteers = policy_scope(User.volunteer).decorate
     @casa_cases = policy_scope(CasaCase.all)
-    @case_contacts = policy_scope(CaseContact.all).order(occurred_at: :desc)
+    @case_contacts = policy_scope(CaseContact.all).order(occurred_at: :desc).decorate
```
Comment (compwron) [inline app/controllers/dashboard_controller.rb:9]: cool :)

File: app/decorators/case_contact_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class CaseContactDecorator < Draper::Decorator
+  delegate_all
+
+  # Returns the contact duration in one of the following formats
```
Comment (compwron) [inline app/decorators/case_contact_decorator.rb:4]: Very cool :)

File: app/models/case_contact.rb [hunk 4]
```diff
@@ -29,18 +29,18 @@ def use_other_type_text?
...
-      minutes_value = duration_minutes.remainder(60)
-
-      return "#{hour_value} #{'hour'.pluralize(hour_value)}" if minutes_value.zero?
-
-      "#{hour_value} #{'hour'.pluralize(hour_value)} #{duration_minutes.remainder(60)} minutes"
-    else
-      "#{duration_minutes} minutes"
-    end
-  end
+  # def display_duration_minutes
```
Comment (compwron) [inline app/models/case_contact.rb:32]: Let's not commit commented-out code where possible

**Comment by GALTdea** [comment]
This was very well documented! thank you @colinsoleim!

**Pull request merged**